torch/distributed/algorithms/ddp_comm_hooks/powerSGD_hook.py

```diff
@@ -30,17 +30,44 @@ def _orthogonalize(matrix, epsilon=1e-8):
 
 
 class PowerSGDState(object):
-    __slots__ = ["process_group", "matrix_approximation_rank", "rng"]
-
-    def __init__(self, process_group, matrix_approximation_rank=1, random_seed=0):
+    __slots__ = [
+        "process_group",
+        "matrix_approximation_rank",
+        "use_error_feedback",
+        "rng",
+        "error_dict",
+    ]
+
+    def __init__(
+        self,
+        process_group,
+        matrix_approximation_rank=1,
+        use_error_feedback=True,
+        random_seed=0,
```

Comment:
Once we formalize the API around powerSGDState, would be great if we could add concrete comuentation for each of these parameters and the tradeoffs involved in using/not using them

Reply:
Let's also make sure to add unit tests either in this Pr or follow up PRs.

Reply:
Let's first prove that PowerSGD can be useful as a comm hook, then we can add more tests and documentations later. Currently the speedup looks promising, but the accuracy loss is a concern, so I hope error feedback can bridge the accuracy gap.